lib/gollum/app.rb

```diff
@@ -583,10 +583,16 @@ def commit_message
     end
 
     def find_upload_dest(path)
-      settings.wiki_options[:allow_uploads] ?
-          (settings.wiki_options[:per_page_uploads] ?
-              "#{path}/#{@name}".sub(/^\/\//, '') : 'uploads'
-          ) : ''
+      if not settings.wiki_options[:allow_uploads]
+        return ''
+      end
+
+      dest = settings.wiki_options[:per_page_uploads] ?
+               "#{path}/#{@name}".sub(/^\/\//, '') : 'uploads'
+      dest = settings.wiki_options[:per_page_uploads_cons] ?
```

Comment:
Don't think the ternary operator is necessary here. Perhaps just:
`dest =  ::File.join([settings.wiki_options['page_file_dir'], 'uploads/', dest].compact) if settings.wiki_options[:per_page_uploads_cons]`

Reply:
Or keep the ternary operator, but ditch the assignment and return statement.
`settings.wiki_options[:per_page_uploads_cons] ? ::File.join([settings.wiki_options['page_file_dir'], 'uploads/', dest].compact) : dest` (and remove the `return` on 595)